torch/lib/ATen/dlpack.h

```diff
@@ -114,6 +114,22 @@ typedef struct {
   /*! \brief The offset in bytes to the beginning pointer to data */
   uint64_t byte_offset;
 } DLTensor;
+
+/*!
+ * \brief C Tensor object, manage memory of DLTensor.
+ */
+typedef struct {
+  /*! \DLTensor which is being memory managed */
+  DLTensor dlTensor;
+  /*! \brief context in which DLManagedTensor is used in a framework */
+  void * ctx;
+  /*! \brief Destructor signature void (*)(void*) - this should be called to
+   *   destruct ctx which holds the DLManagedTensor. It can be NULL if there is
+   *   no way for the caller to provide a reasonable destructor.
+   */
+  void (*destructor)(void * ctx) = NULL;
```

Comment:
The destructor needs both arguments it had before because other implementations may use them (even if the PyTorch implementation does not need it). This is a C idiom that allows for flexible callbacks:

```
void (*destructor)(DLManagedTensor * self, void * ctx); 
```
 `= NULL;` is a C++ism so it can't go in this C-only.

Reply:
I see, I was wondering about this and also checked the C2 MemoryDeleter which also has the signature `void (*)(void *)`

Reply:
`= NULL` seems to be a valid thing to do in C as well https://www.tutorialspoint.com/cprogramming/c_pointers.htm and I also verified it in the code by not setting the destructor locally and it seems to be fine but perhaps I am missing something obvious.

Reply:
It's not valid in C. Try running this through your C compiler, e.g., `gcc -c foo.c`:

```
typedef struct {
    int (*get_int)() = NULL;
} foo;
int main() {
    return 0;
}
```